Update build_from_source (#2272)
openvinotoolkit/model_server: docs/build_from_source.md

## Makefile and building

Makefile located in root directory of this repository contains all targets needed to build docker images and binary packages. One of these targets is `release_image` which by default builds `openvino/model_server:latest` image.

```bash
git clone https://github.com/openvinotoolkit/model_server
cd model_server
make release_image
````

> To build image with support of additional plugins use parameters described in paragraphs below.

Reviewer: ```suggestion
To build the image with non default configuration, add parameters described below.
```

Example:
```bash
make release_image MEDIAPIPE_DISABLE=1
```

### `PYTHON_DISABLE`

> When set to `0`, OpenVINO&trade Model Server will be built with [Python Nodes](python_support/quickstart.md) support. Default value: `0`. 

Reviewer 2: Can we say what happens when set to `1`? It feels more natural imho,  since it's basically a flag. 
We could do the same for `MEDIAPIPE_DISABLE`

Example:
```bash

> make release_image PYTHON_DISABLE=0

Reviewer: ```suggestion
make release_image PYTHON_DISABLE=1
```

```

> 

Reviewer 2: Maybe add note that Python is enabled via MediaPipe and configuration with `MEDIAPIPE_DISABLE=1 PYTHON_DISABLE=0` is invalid

### `GPU`

> When set to `1`, OpenVINO&trade Model Server will be built with [GPU plugin](https://docs.openvino.ai/2023.2/openvino_docs_OV_UG_supported_plugins_GPU.html) support. Default value: `0`.

Reviewer: ```suggestion
When set to `1`, OpenVINO&trade Model Server will be built with the drivers required by [GPU plugin](https://docs.openvino.ai/2023.2/openvino_docs_OV_UG_supported_plugins_GPU.html) support. Default value: `0`.
```

Example:
```bash
make release_image GPU=1
```

## Building minimal image

> Building minimalistic OVMS docker image requires disabling all additional features:

Reviewer: ```suggestion
Building minimalistic OVMS docker image requires disabling all optional features:
```

```bash
make release_image GPU=0 MEDIAPIPE_DISABLE=1 PYTHON_DISABLE=1 SENTENCEPIECE=0
```

## Building Binary Package

> Building ovms.tar.gz package is possible by using targz_package target:

Reviewer: ```suggestion
The binary packages includes the `ovms` executable and linked libraries for bare metal deployments. It includes also a shared library for the model server CAPI interface. Building `ovms.tar.gz` package is possible by using `targz_package` target:
```

```bash

> make targz_package

Reviewer: ```suggestion
make targz_package PYTHON_DISABLE=1
```
